src/generators/go/renderers/StructRenderer.ts

```diff
@@ -0,0 +1,42 @@
+import { GoRenderer } from '../GoRenderer';
+import { StructPreset } from '../GoPreset';
+import { FormatHelpers } from '../../../helpers/FormatHelpers';
+import { pascalCaseTransformMerge } from 'pascal-case';
+
+/**
+ * Renderer for Go's `struct` type
+ * 
+ * @extends GoRenderer
+ */
+export class StructRenderer extends GoRenderer {
+  public async defaultSelf(): Promise<string> {
+    const content = [
+      await this.renderFields(),
+      await this.runAdditionalContentPreset()
+    ];
+
+    const formattedName = this.model.$id && FormatHelpers.toPascalCase(this.model.$id, { transform: pascalCaseTransformMerge });
+    return `// ${formattedName} represents a ${formattedName} model.
+type ${formattedName} struct {
+${this.indent(this.renderBlock(content, 2))}
+}`;
+  }
+}
+
+export const GO_DEFAULT_STRUCT_PRESET: StructPreset<StructRenderer> = {
+  self({ renderer }) {
+    return renderer.defaultSelf();
+  },
+  ctor() {
+    return 'thisShoulBeAConstructor';
+  },
+  field({ fieldName, field, renderer }) {
+    return `${FormatHelpers.toPascalCase(fieldName, { transform: pascalCaseTransformMerge }) } ${ renderer.renderType(field)}`;
+  },
+  getter() {
+    return 'getterFunc';
+  },
+  setter() {
+    return 'setterFunc';
+  },
```

Comment:
I have in mind these parts when I wrote in another comments:

> This is important, as I see in the other parts of PR, some code is still in development, so maybe we should not export at the moment new language? :)

So we shouldn't export at the moment the Golang generator, or... please write the implementation of `setter`, `getter`, `ctor` 😄

Reply:
I understand that merging code that is not complete and making it available to the final user could lead to some misunderstanding, believing that we support Go in Modelina. However, progressive changes are always better. 
You suggest to skip the export part, but then tests won't work ever as they need to import the Generator, among others. Any smart way of fixing this?

Reply:
You can import them from path where they are:

```ts
import ... from ".../src/generators/go/renderers/StructRenderer.ts"
```

etc.

Reply:
Ok, I removed the `export * from './GoGenerator';` from the `index.ts` file so `GoGenerator` is un-exported now.

Reply:
Why not just remove the hooks so the Go generator just generates a simple struct with fields, no accessors or constructor? 

I feel like it is wrong to merge something that cannot be used unless there is a specific reason/plan to enable the support 😄

Reply:
If we do that we would then need to add proper documentation mentioning the limitations and the current state of the generator (pretty early days, limited features).
I'm not against that, In fact I see it can add value to the final user. Some users will find this, generation of structs, useful for their projects.
If we move with that option, I vote for merging this PR, then working on enabling it in another PR so we do not pollute this one and have to review it again. 

Tbh, I vote to make the generator public so users can use it (as @jonaslagoni suggested).

Reply:
> If we do that we would then need to add proper documentation mentioning the limitations and the current state of the generator (pretty early days, limited features).

To be fair to me as long as the repo is < v1 all generators are to be seen as WIP, but you right we should probably start defining the states of the generators 🤔 

> If we move with that option, I vote for merging this PR, then working on enabling it in another PR so we do not pollute this one and have to review it again.

That's fair, that works for me 👍 

@magicmatatjahu any thoughts?

Reply:
Ok, merge this and work on another :)